### PR TITLE
Implemented search on api and provided a client assembly.

### PR DESCRIPTION
**Pull request by MichaelYoung1981**

### CODE REVIEW

**Comment by EwanNoble** [comment]
Are all 17 of those cloud service settings needed? 😄

File: src/SFA.DAS.EmployerUsers.Api/Controllers/SearchController.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using SFA.DAS.EmployerUsers.Api.Orchestrators;
+
+namespace SFA.DAS.EmployerUsers.Api.Controllers
+{
+    [RoutePrefix("api/users/search")]
+    public class SearchController : ApiController
+    {
+        private readonly UserOrchestrator _orchestrator;
+
+        public SearchController(UserOrchestrator orchestrator)
```
Comment (simonbu11) [inline src/SFA.DAS.EmployerUsers.Api/Controllers/SearchController.cs]: I am not sure orchestrators are adding any value here, however if we are going to us them then we should follow the 1:1 controller:orchestrator pattern - https://www.simple-talk.com/dotnet/asp-net/never-mind-the-controller-here-is-the-orchestrator/

**Comment by simonbu11** [comment]
The search controller seems a bit odd to me. Search is not really a resource. I would of expected /api/user?filter=somestuff personally. Though this is possibly a style thing

**Comment by floydprice** [comment]
I agree that a search endpoint is not strictly Restful but its a common pattern to have a dedicated search end point. I'm comfortable with this approach for now but will raise with the TA clan.

**Comment by floydprice** [comment]
@simonbu11 please review the changes relating to https://github.com/SkillsFundingAgency/das-employerusers/pull/28#pullrequestreview-21753943

**Pull request merged**